popup.js

```diff
@@ -12,19 +12,96 @@ async function listAllSVGElements() {
 
     let list = document.getElementById('svgList');
     for (const item of response[0].result) {
+        //generate a list element for each svg found
         let li = document.createElement('li');
-        li.innerHTML = item;
-        var downloadLink = generateDownloadLink(item);
-        let button = document.createElement('button');
-        button.innerHTML = "Download";
-        li.appendChild(button);
+        li.id = getRandomSvgName();
+
+        //add checkbox
+        let checkbox = document.createElement('input');
+        checkbox.type = 'checkbox';
+        checkbox.value = 'unchecked';
+        li.appendChild(checkbox);
+        checkbox.addEventListener('input', onCheckboxInput);
+
+        //add svg display
+        let svgItem = document.createElement('svg');
+        li.appendChild(svgItem);
+        svgItem.outerHTML = item;
+
+        //TO DO: get dimension info from the svg
+
+        //TO DO: add or update SVG viewBox attribute to prevent overflow
+
+        let infoContainer = document.createElement('div');
+        infoContainer.classList.add('info-container');
+        li.appendChild(infoContainer);
+
+        //add name display
+        let nameContainer = document.createElement('div');
+        nameContainer.classList.add('name-container');
+        infoContainer.appendChild(nameContainer);
+
+        let nameLabel = document.createElement('span');
+        nameLabel.innerHTML = 'Name:';
+        nameContainer.appendChild(nameLabel);
+        let nameInput = document.createElement('input');
+        nameInput.type = 'text';
+        nameInput.id = li.id + '-name';
+        nameInput.value = li.id;
+        nameContainer.appendChild(nameInput);
+
+        //update file name on name input change (only fired on focus lost)
+        nameInput.addEventListener('change', updateSvgName);
```

Comment:
Change olduğu için ilk seferde de updateSvgName fonksiyonunu çağırıyor olabilir mi? 
<img width="618" alt="Screen Shot 2022-05-05 at 15 29 44" src="https://user-images.githubusercontent.com/67782100/166923267-9d72ae36-30a4-447b-9d9d-0c2566ccdfdd.png">

Reply:
İnceledim, ilk indirmeden sonra tekrar indirmemesi ile aynı sebepten. İlkinden sonra bu downloadLink elemanı kayboluyor, sonrasında dosya ismini değiştirmeye çalışınca hatayı veriyor.